Mage.Sets/src/mage/cards/s/SanctumOfAll.java

```diff
@@ -0,0 +1,174 @@
+
+        // If an ability of another Shrine you control triggers while you control six or more Shrines, that ability triggers an additional time.
+        this.addAbility(new SimpleStaticAbility(new SanctumOfAllTriggerEffect()).addHint(new ValueHint("Shrines you control", count)));
+    }
+
+    private SanctumOfAll(final SanctumOfAll card) {
+        super(card);
+    }
+
+    @Override
+    public SanctumOfAll copy() {
+        return new SanctumOfAll(this);
+    }
+}
+
+class SanctumOfAllSearchEffect extends OneShotEffect {
+
+    public static final FilterCard filter = new FilterCard("a Shrine card");
+
+    static {
+        filter.add(SubType.SHRINE.getPredicate());
+    }
+
+    SanctumOfAllSearchEffect() {
+        super(Outcome.Benefit);
+        staticText = "search your library and/or graveyard for a Shrine card and put it onto the battlefield. If you search your library this way, shuffle it";
+    }
+
+    private SanctumOfAllSearchEffect(SanctumOfAllSearchEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public SanctumOfAllSearchEffect copy() {
+        return new SanctumOfAllSearchEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null) {
+            Optional<Card> maybeChosenCard = Optional.empty();
+            boolean searchedLibrary = false;
+
+            // from library
+            Target target = new TargetCardInLibrary(0, 1, filter).withChooseHint("from library");
+            if (target.canChoose(source.getSourceId(), controller.getId(), game)
+                    && controller.chooseUse(Outcome.Benefit, "Search your library?", source, game)
+                    && target.choose(Outcome.PutCardInPlay, controller.getId(), source.getSourceId(), game)) {
+                maybeChosenCard = Optional.ofNullable(game.getCard(target.getFirstTarget()));
+                searchedLibrary = true;
+            }
+
+            if (!maybeChosenCard.isPresent()) {
+                // from graveyard
+                target = new TargetCardInYourGraveyard(0, 1, filter, true).withChooseHint("from graveyard");
+                if (target.canChoose(source.getSourceId(), controller.getId(), game)
+                        && target.choose(Outcome.PutCardInPlay, controller.getId(), source.getSourceId(), game)) {
+                    maybeChosenCard = Optional.ofNullable(game.getCard(target.getFirstTarget()));
+                }
+            }
+
+            maybeChosenCard.ifPresent(card -> controller.moveCards(card, Zone.BATTLEFIELD, source, game));
+            if (searchedLibrary) {
+                controller.shuffleLibrary(source, game);
+            }
+            return true;
+        }
+        return false;
+    }
+}
+
+class SanctumOfAllTriggerEffect extends ReplacementEffectImpl {
+
+    private static final FilterPermanent filter = new FilterPermanent();
```

Comment:
Miss controlled filter

Reply:
actually didn't need that filter anymore so removed it